### PR TITLE
Run syntax tests when grammar changes

### PR DESCRIPTION
**Pull request by jpogran**
This runs the syntax tests only if the grammar files change in the syntaxes directory.

As we start to modify the syntax and grammar more, we'll be running tests more frequently. We don't need to incur the cost or time of running the entire suite when we are testing grammar changes.

### CODE REVIEW

File: .github/workflows/syntax.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Run syntax tests
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'syntaxes/**'
```
Comment (dbanck) [inline .github/workflows/syntax.yml:8]: We might also want to run this when the syntax test files change: `'tests/**'`

**Review by dbanck — CHANGES_REQUESTED** [review]
To be sure: this will still run the main `test.yml` workflow on each syntax change, as long as we don't exclude the syntax paths in `test.yml`, right?

**Comment by jpogran** [comment]
Good point. This didn't go far enough and prevent `test.yml` from running when there are just syntax changes

**Review by dbanck — APPROVED** [review]
LGTM!

**Pull request merged**